### PR TITLE
Fix _dd.p.ksr scientific notation for very small sampling rates

### PR DESCRIPTION
**Pull request by bm1549**
### Description

Fix `_dd.p.ksr` span tag formatting for very small sampling rates. Rates below 0.001 were formatted using C's `%.6g` format which outputs scientific notation (e.g. `0.000001` → `"1e-06"`). Changed to explicit integer-level rounding via `floor(rate * 1e6 + 0.5)` and `%.6f` formatting to always produce decimal notation with up to 6 decimal digits, trailing zeros stripped.

Fixes APMAPI-1869

**Related PRs:**
- dd-trace-py: https://github.com/DataDog/dd-trace-py/pull/17086
- dd-trace-rb: https://github.com/DataDog/dd-trace-rb/pull/5497
- dd-trace-js: https://github.com/DataDog/dd-trace-js/pull/7846
- system-tests: https://github.com/DataDog/system-tests/pull/6466

### Testing

- Updated test description in `tests/ext/priority_sampling/028-ksr-tag-formatting.phpt`
- Added new test `tests/ext/priority_sampling/029-ksr-tag-small-rate-formatting.phpt` verifying:
  - Rate 0.000001 → `"0.000001"` (decimal notation, not `"1e-06"`)
  - No scientific notation characters in output

### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: ext/priority_sampling/priority_sampling.c [hunk 1]
```diff
@@ -65,8 +66,20 @@ static void dd_update_decision_maker_tag(ddtrace_root_span_data *root_span,
 static void dd_update_knuth_sampling_rate_tag(ddtrace_root_span_data *root_span, double sample_rate) {
     zend_array *meta = ddtrace_property_array(&root_span->property_meta);
 
+    // Round to 6 decimal places at integer level to avoid IEEE 754 precision issues,
+    // then format with fixed-point notation (never scientific notation).
+    double rounded = floor(sample_rate * 1e6 + 0.5) / 1e6;
```
Comment (bwoebi) [inline ext/priority_sampling/priority_sampling.c:71]: Are you sure?
snprintf() should round itself.

Doing this should covert `0.0000001` to `0.000001` instead of `0`

Comment (bwoebi) [reply ext/priority_sampling/priority_sampling.c:71]: ~Or maybe that's just PHP and not snprintf.~ - a quick google lets me find https://stackoverflow.com/a/24120950.

Comment (bm1549) [reply ext/priority_sampling/priority_sampling.c:71]: 0.0000001 (1e-7) should go to 0 since we only want 6 decimal places

Comment (bwoebi) [reply ext/priority_sampling/priority_sampling.c:71]: Ah, but the rounding is implementation defined, so actually libc defined.

Comment (bwoebi) [reply ext/priority_sampling/priority_sampling.c:71]: Ah, that's why floor(). Got it.

Comment (bm1549) [reply ext/priority_sampling/priority_sampling.c:71]: The test cases are chosen deliberately to highlight the edge cases I wanted to cover

```
0.000001   -> 0.000001
0.00000051 -> 0.000001
0.0000001  -> 0

Comment (bwoebi) [reply ext/priority_sampling/priority_sampling.c:71]: Yes, I just was sort of missing the floor and just seeing the 0.5, and confused. You could have used ap_php_snprintf(), which has a fixed behaviour and you use the rounding as described there, but it's good as is.
